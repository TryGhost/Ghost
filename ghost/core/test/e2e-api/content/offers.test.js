const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const testUtils = require('../../utils');
const models = require('../../../core/server/models');

const offerSnapshot = {
    id: matchers.anyObjectId,
    tier: {
        id: matchers.anyObjectId
    }
};

describe('Offers Content API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getContentAPIAgent();
        await fixtureManager.init('api_keys', 'members');
        await agent.authenticate();
    });

    it('Can read offer details from id', async function () {
        const productModel = await models.Product.findOne({type: 'paid'}, testUtils.context.internal);

        const offerData = testUtils.DataGenerator.forKnex.createOffer({
            product_id: productModel.get('id')
        });
        const offerModel = await models.Offer.add(offerData, {context: {internal: true}});

        await agent.get(`/offers/${offerModel.get('id')}`)
            .expectStatus(200)
            .matchHeaderSnapshot({
                'content-version': matchers.anyContentVersion,
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                offers: Array(1).fill(offerSnapshot)
            });
    });
});
