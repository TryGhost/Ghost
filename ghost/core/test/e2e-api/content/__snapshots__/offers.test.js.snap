// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers Content API Can read offer details from id 1: [body] 1`] = `
Object {
  "offers": Array [
    Object {
      "amount": 12,
      "cadence": "year",
      "currency": null,
      "currency_restriction": false,
      "display_description": "",
      "display_title": "",
      "duration": "once",
      "duration_in_months": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Black Friday",
      "status": "active",
      "tier": Object {
        "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
        "name": "Default Product",
      },
      "type": "percent",
    },
  ],
}
`;

exports[`Offers Content API Can read offer details from id 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "public, max-age=0",
  "content-length": "331",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
